extensions/amp-viewer-integration/0.1/amp-viewer-integration.js

```diff
@@ -111,10 +112,26 @@ export class AmpViewerIntegration {
         }
       );
     }
-    /** @type {?HighlightInfoDef} */
-    const highlightInfo = getHighlightParam(ampdoc);
-    if (highlightInfo) {
-      this.highlightHandler_ = new HighlightHandler(ampdoc, highlightInfo);
+
+    if (
+      'fragmentDirective' in document &&
+      isExperimentOn(this.win, 'enable-text-fragments')
+    ) {
+      this.win.addEventListener('message', (e) => {
+        if (
+          e.origin !== 'https://www.google.com' ||
```

Comment:
Is this checking if we're from a proxy origin?

Reply:
Talked offline, this is checking for a message from GWS. Maybe add a comment here. 


Also we are only supporting text fragments from GWS?

Reply:
Yes, we're only supporting text fragments from GWS since we're appending it to the URL.

Reply:
Can this origin be pluggable from configuration? There are other AMP Viewer users who might find this feature an improvement over the current.

Reply:
Not just that, there are other Google domains that can use this. Not all SERPs are `google.com`.

Reply:
How about a pluggable allow list from a configuration? Checking event origin is important to prevent cross-site scripting attacks.

Reply:
I don't think there's any reason to restrict this, especially given the fallback is available to everyone currently.

Reply:
Outdated comment since implementation changed to using `highlight` param.